app/uk/gov/hmrc/slacknotifications/model/NotificationRequest.scala

```diff
@@ -21,25 +21,29 @@ import play.api.libs.json.Reads._
 import play.api.libs.json.{Reads, _}
 
 final case class MessageDetails(
-                                 text: String,
-                                 username: String,
-                                 iconEmoji: Option[String] = None,
-                                 attachments: Seq[Attachment] = Nil
-                               )
+  text: String,
+  username: String,
+  iconEmoji: Option[String]    = None,
+  attachments: Seq[Attachment] = Nil
+) {
+
+  def getFields: Array[String] = Array(text, username, iconEmoji.getOrElse(""))
```

Comment:
There's a lot of usage of Array which is mutable - prefer Vector or List.

Reply:
I've made this change